pkg/controller/rollout/rollout_canary.go

```diff
@@ -454,6 +454,11 @@ func (m *canaryReleaseManager) finalizingBatchRelease(c *RolloutContext) (bool,
 		return false, err
 	}
 	klog.Infof("rollout(%s/%s) patch batchRelease(%s) success", c.Rollout.Namespace, c.Rollout.Name, body)
+
+	// if rollout is disabling, then the batchrelease should be deleted
+	if c.NewStatus.Phase == v1alpha1.RolloutPhaseDisabling {
```

Comment:
Remove the code, because the progressing will never run this, when rollout.spec.disabled is true.